src/main/java/com/aws/greengrass/clientdevices/auth/configuration/GroupConfiguration.java

```diff
@@ -114,4 +107,21 @@ private Set<String> findPolicyVariables(String resource) {
         }
         return policyVariables;
     }
+
+    /**
+     * Validate the deviceGroups configuration.
+     *
+     * @throws PolicyException if an invalid policy is detected
+     */
+    public void validate() throws PolicyException {
```

Comment:
Should we also validate that policy variables in policy document are supported by the `PolicyVariableResolver` since the resolver doesn't handle unsupported policy variables?

Reply:
yep, wanted to do that in a follow-up, this PR is meant to be a refactor